packages/core/src/NodeExecuteFunctions.ts

```diff
@@ -1647,6 +1652,187 @@ function cleanupParameterData(inputData: NodeParameterValueType): NodeParameterV
 	return inputData;
 }
 
+function findPropertyInArray(
```

Comment:
1. Value extraction with `extractValue` relies on five other functions: `findPropertyFromParameterName`, `findPropertyInArray`, `extractValueRLC`, `executeRegexExtractValue`, and `extractValueOther`. Seeing as `NodeExecuteFunctions.ts` is already quite large, would it be reasonable to move all value extraction logic to a module that exports `extractValue`?
2. Is it really worth doing all this setup only for `options.find((i) => i.name === name);`? I would remove the helper and call `find` directly: `property = property.options?.find((i) => i.name === param);` Same for `values`.

Reply:
1. They could definitely be moved into their own file.
2. This is sort of a hold over from an earlier iteration. I kept it because I wasn't sure if I'd make any changes to how it worked. There was also the possibility of returning multiple results so it ended up staying.